### ISSUE TITLE
Millions of duplicates in images database table

### ISSUE THREAD

**Issue by grololo06** [ecotaxa/ecotaxa_dev#583]
Images from the same origin, for the _same object_ (so not in a subset) , but different lines:

```
 select count(*) from images img
 where exists(select 1 from images img2 where 
              img2.objid=img.objid -- same object
              and img2.orig_file_name=img.orig_file_name  -- same origin file
              and img2.imgid != img.imgid -- different lines in table
) 

```
6 495 629

I guess the TSV update was wrong at some point.
I compared a couple of them and they are identical.
TODO:

- [x] Issue a list of the remove-able rows.
- [x] Remove from DB
- [x] Remove from filesystem.

**Comment by grololo06** [comment]
Generator for cleanup script:
```
select 'diff '||img.file_name||' '||img2.file_name||' && rm '||img2.file_name||' && psql -U postgres -d ecotaxa -h localhost -c "delete from images where imgid='||img2.imgid||'"'
 from images img, images img2
 where img2.objid=img.objid -- same object
   and img2.orig_file_name=img.orig_file_name  -- same origin file
   and img2.width=img.width
   and img2.height=img.height
   and img2.imgid > img.imgid
```

**Comment by grololo06** [comment]
After failed attempts to reproduce in local environment, it looks like there is nothing better to do than cleaning-up and waiting for another occurrence, if ever.

**Comment by grololo06** [comment]
I decided to integrate the cleanup into the application code, as duplicate image detection will be useful for all pending issues on image management.

**Comment by grololo06** [comment]
Impacted projects:
```
create temp table objs_with_dup as select distinct img.objid
from images img, images img2
 where img2.objid=img.objid -- same object
   and img2.orig_file_name=img.orig_file_name  -- same origin file
   and img2.width=img.width
   and img2.height=img.height
   and img2.imgid > img.imgid
   
select projid, count(1) as cnt from objects 
where objid in (select objid from objs_with_dup)
group by projid
order by cnt desc
```

**Comment by grololo06** [comment]
```
627	952963
2485	615552
43	389101
3660	385890
3186	255358
3539	80774
3538	73632
3179	47536
3580	45411
3953	45312
264	44292
3894	32935
3203	28391
517	26394
3290	19148
2717	18804
3701	17026
2711	9046
2979	3492
3114	3407
3134	2622
2337	2207
3857	1922
1596	1878
3902	1286
3995	1133
2304	1124
2941	1013
529	685
552	685
3209	580
3201	580
3202	579
3205	579
3204	578
3208	576
3786	557
3177	417
3594	389
3217	162
3196	65
3194	65
3198	65
3193	65
3197	65
3199	64
3189	64
3192	64
3195	63
4012	13
3518	11
3852	8
3374	5
```

**Comment by grololo06** [comment]
Can be reproduced easily by re-importing the same file with the option "Skip objects that have already been imported " :(

**Comment by grololo06** [comment]
Fixed in 2.5.6
